### ISSUE TITLE
Forgot your password: Should we bring users to the old screen?

### ISSUE THREAD

**Issue by escapemanuele** [Automattic/wp-calypso#96291]
### Quick summary

![Image](https://github.com/user-attachments/assets/63855fcc-163d-4f5a-9134-931bacc6f50e)

Link at the bottom brings the user to an older page

https://wordpress.com/wp-login.php?action=lostpassword

While "Forgot password?" to https://wordpress.com/log-in/lostpassword, that even if not beautiful is in Calypso.

### Steps to reproduce

- Go to login page

### What you expected to happen

Decide where to send the user if they forgot the password

### What actually happened

Two different links

### Impact

All

### Available workarounds?

There is no user impact

**Comment by gavande1** [comment]
I don't think so. We should keep the experience consistent. Honestly, it seems like a mistake to me. Let me check if it's intentional.

**Comment by escapemanuele** [comment]
Yeah 100% a mistake as it should be consistent. I'm not sure of the status of the Calypso lost your password page, as I've not worked on that. Maybe @chihsuan knows more

**Comment by gavande1** [comment]
It appears that the Calypso screen is not fully functional. After submitting my email, it redirected me to log in without providing any message. I did receive the email, but the link directed me to the old UI (wp-login.php). I have a feeling that keeping the old UI may have been a conscious decision.

**Comment by escapemanuele** [comment]
> I have a feeling that keeping the old UI may have been a conscious decision.

Yeah, that may be it. We should probably keep the old UI for now, but strive to get to the new version, as the jump from one screen to the other is a bit strange.

**Comment by gavande1** [comment]
Perfect. I will prepare a fix for that. Instead of redirecting users to the Calypso screen to reset their passwords, it will redirect them to the Old UI for consistency.

**Comment by gavande1** [comment]
👋 @chihsuan, I am restoring the changes introduced in the https://github.com/Automattic/wp-calypso/pull/68699 PR, which introduces a react-based lost password screen. Before making changes to it, I wanted to confirm with you that the changes are restored and if you have any concerns with that. 

I was thinking of making changes to [renderLostPasswordLink()](https://github.com/Automattic/wp-calypso/blob/cff770810c9064424ef818c12033b166da7f8ba0/client/blocks/login/login-form.jsx#L623-L623).  Let me know what you think about it. 

@escapemanuele, Originally, we used the old lost password screen. Then, the react-based screen was introduced behind the `login/react-lost-password-screen` feature, and the flag was enabled for all environments.

**Comment by chihsuan** [comment]
> I am restoring the changes introduced in the https://github.com/Automattic/wp-calypso/pull/68699 PR, which introduces a react-based lost password screen. Before making changes to it, I wanted to confirm with you that the changes are restored and if you have any concerns with that.

@gavande1 When I introduced the react-based lost password screen, I made sure that it was enabled for WooCommerce.com only. I am not sure why it was enabled for all environments. As long as the feature is enabled for Woo, I don't have any concerns. Thanks for checking in with me! 🙏

**Comment by escapemanuele** [comment]
I'd love to have the react screen, but @chihsuan told me that it's tested on Woo. So, I believe it'll be necessary to do some testing and maybe tweak things a little. Hopefully, not much.

The react screen also needs some styling work, but it's almost there.

**Comment by gavande1** [comment]
Idea: How about we take this up as a mini project and revamp the forgot password screen and standardize the calypso screen?

**Comment by gavande1** [comment]
I will mark this issue's status as `Need Shaping`.

**Comment by escapemanuele** [comment]
I feel like there is some need for that, as it is now a really neglected part. Thanks

**Comment by StevenDufresne** [comment]
> @gavande1 When I introduced the react-based lost password screen, **I made sure that it was enabled for WooCommerce.com only**. I am not sure why it was enabled for all environments. As long as the feature is enabled for Woo, I don't have any concerns. Thanks for checking in with me! 🙏

@chihsuan I believe that is only partly true. It appears like the [original PR](https://github.com/Automattic/wp-calypso/pull/68885) introduced a defect that was sorted in #96506. Additionally, looking at that PR, there isn't any `isWoo` checks for [rendering the react view](https://github.com/Automattic/wp-calypso/pull/68885/files#diff-a2085484a90c3affa18d524a40f82a2dcd6f62819f3c9246abdb4dca23feb5e2R277) itself and [the route is added for all cases](https://github.com/Automattic/wp-calypso/pull/68885/files#diff-8b66011588360680c38b6923b3207d38dec4282037d8adbd40ea3dd2c68262a5R102).

Since there are no longer any links to the React view accessible from dotCOM, and the [dotCom knowledge base uses the PHP lost-password](https://wordpress.com/support/passwords/), I would suggest we redirect wp.com users to `wp-login.php?action=lostpassword`. 

I can look into that.

**Comment by chihsuan** [comment]
> I  believe that is only partly true. It appears like the https://github.com/Automattic/wp-calypso/pull/68885 introduced a defect that was sorted in https://github.com/Automattic/wp-calypso/pull/96506. Additionally, looking at that PR, there isn't any isWoo checks for [rendering the react view](https://github.com/Automattic/wp-calypso/pull/68885/files#diff-a2085484a90c3affa18d524a40f82a2dcd6f62819f3c9246abdb4dca23feb5e2R277) itself and [the route is added for all cases](https://github.com/Automattic/wp-calypso/pull/68885/files#diff-8b66011588360680c38b6923b3207d38dec4282037d8adbd40ea3dd2c68262a5R102).

@StevenDufresne The defect logic was introduced by another PR. The original PR [only rendered the lost password link for Woo](https://github.com/Automattic/wp-calypso/pull/68885/files#diff-c4b202c48b075713a04477b1013b69e705758b4a0db59b3e648a48ed95077eddR727). It's intentional that there is no isWoo check for the route since the lost password link (action=lostpassword param) was only available for WooCommerce.com at that time. 

> Since there are no longer any links to the React view accessible from dotCOM, and the dotCom knowledge base uses the PHP lost-password, I would suggest we redirect wp.com users to wp-login.php?action=lostpassword or alternatively remove the route.

That sounds good.

**Comment by StevenDufresne** [comment]
Ah [you're right](https://github.com/Automattic/wp-calypso/commit/0425c28813a1d8873e6c104c75cb245c55766142#diff-c4b202c48b075713a04477b1013b69e705758b4a0db59b3e648a48ed95077eddR937), thanks for the follow-up!
